aiohttp/abc.py

```diff
@@ -139,6 +139,10 @@ async def close(self) -> None:
 class AbstractCookieJar(Sized, IterableBase):
     """Abstract Cookie Jar."""
 
+    @abstractmethod
+    def clone(self) -> "AbstractCookieJar":
```

Comment:
Return type would need to be a TypeVar, this should return the same class, rather than AbstractCookieJar.
e.g. https://mypy.readthedocs.io/en/stable/cheat_sheet_py3.html?highlight=TypeVar#decorators

Reply:
could you explain more about this? like this?

```python
CookieJar = TypeVar("CookieJar", bound="AbstractCookieJar")
def clone(self: CookieJar) -> Type[CookieJar]:
```

Reply:
Looks about right. I'd name it `_CookieJar` to keep in private.

Reply:
I have updated my commit, please review when got time